packages/config/config/devices/0x0330/VES-ZW-MOT-018.json

```
@@ -0,0 +1,248 @@
+{
+	"manufacturer": "Vesternet",
+	"manufacturerId": "0x0330",
+	"label": "VES-ZW-MOT-018",
+	"description": "Motor Controller",
+	"devices": [
+		{
+			"productType": "0x0004",
+			"productId": "0xd00d",
+			"zwaveAllianceId": 3342
+		}
+	],
+	"firmwareVersion": {
+		"min": "0.0",
+		"max": "255.255"
+	},
+	"paramInformation": [
+		{
+			"#": "2",
+			"label": "Send Multi Level Report On State Change",
+			"description": "Report percentage when level changes, unit is 1%",
+			"unit": "1%",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 10,
+			"unsigned": true,
+			"defaultValue": 5,
+			"options": [
+				{
+					"label": "Disable",
+					"value": 0
+				}
+			]
+		},
+		{
+			"#": "3",
+			"label": "Working Mode",
+			"description": "Change mode of device",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 2,
+			"unsigned": true,
+			"defaultValue": 2,
+			"options": [
+				{
+					"label": "Light",
+					"value": 0
+				},
+				{
+					"label": "Shutter Mode Without Positioning",
+					"value": 1
+				},
+				{
+					"label": "Shutter Mode With Positioning",
+					"value": 2
+				}
+			]
+		},
+		{
+			"#": "4",
+			"label": "Enable / Disable Save Positioning Percentage",
+			"$import": "~/templates/master_template.json#base_enable_disable",
+			"defaultValue": 1
+		},
+		{
+			"#": "5",
+			"label": "Over Current Protection",
+			"description": "Triggered when module goes over current limit.",
+			"$import": "~/templates/master_template.json#base_enable_disable",
+			"defaultValue": 1
+		},
+		{
+			"#": "6",
+			"label": "Start Positioning Calibration",
+			"description": "ONLY VALID IN SHUTTER WITH POSITIONING MODE. Enable to begin, will revert to Disabled once completed.",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 1,
+			"unsigned": true,
+			"defaultValue": 0,
+			"options": [
+				{
+					"label": "Not Calibrated",
+					"value": 0
+				},
+				{
+					"label": "Calibrated",
+					"value": 1
+				}
+			]
```

I suppose this would make more sense given the above description
```suggestion
			"options": [
				{
					"label": "Idle / Not calibrated",
					"value": 0
				},
				{
					"label": "Calibrate",
					"value": 1
				}
			]
```

---

This one's a bit tricky, you can only write value 1 to the Parameter, to start the calibration.  

But it reads back as either 0 (not yet calibrated, or failed to calibrate) or 1 (calibrated).

Not sure how best to handle that since 1 can mean two different things, start or completed?

Maybe:

```
"description":  "Only valid in Shutter Mode with positioning. Set to 'Calibrate / Calibrated' to begin, will show 'Not Calibrated' or 'Calibrate / Calibrated' once complete."

		"options": [
			{
				"label": "Not Calibrated",
				"value": 0
			},
			{
				"label": "Calibrate / Calibrated",
				"value": 1
			}
		]
```

---

I noticed the availability of conditional parameters - https://github.com/zwave-js/node-zwave-js/blob/master/docs/config-files/conditional-settings.md

I don't suppose it's possible to use the value of another parameter as a condition to change / enable / hide other parameters?

So for example with this calibration parameter, if the current value is 0 then it could show the label for the 2nd option as "start calibration" and if the current value is 1 then it could show it as "calibrated".

It would also be useful to be able to hide parameters that are only applicable when the device is configured in a specific mode, e.g. those that say "only valid in light mode" to only be shown when that's true.

---

Sadly we can't correlate config parameters in that way yet. I've opened an issue for that, we have some other use case too: https://github.com/zwave-js/node-zwave-js/issues/5583

About the option values: let's take your suggestion. I might have misunderstood the original description.

---

Ah yes, I was trying to rack my brain to remember what other devices had similar dynamic parameters.  I think some Qubino and Secure (Horstmann) do.